### ISSUE TITLE
Copy-implementing variables are flagged as `Use of moved value`

### ISSUE THREAD

**Issue by David-OConnor** [intellij-rust/intellij-rust#5691]
## Environment

* **IntelliJ Rust plugin version:** 0.2.114.2156-192
* **Rust toolchain version:** 0.2.125.3191-201
* **IDE name and version:** PyCharm 2020.1.2
* **Operating system:** Windows 10 x64 Professional

## Problem description
Copy-implementing variables are flagged as `Use of moved value`

## Steps to reproduce

```rust
#[derive(Clone, Copy)]
pub enum A {
    B,
    C,
}

fn use(v: A) {
}

fn d() {
    let e = A::B;
    use(e);
    use(e);  // triggers red underline error, and red tick on the right
}
-->

**Comment by nazar-pc** [comment]
I hit this in all projects starting with simple variables and finishing with structs having `u32` fields for instance. In many cases this is the only reason the whole file is marked red.

**Comment by nazar-pc** [comment]
Probably a duplicate of #4361, #4858, #3840, #3987 and #3175.
Looks like someone needs to collapse all of these.

**Comment by ortem** [comment]
@nazar-pc It is not a duplicate because the causes are different: lack of `no_std` support, invalid caches, wrong name resolution, and so on. Please create a new issue (you can do it right from IDE using `Help | Create New Issue` action) if your projects don't use `no_std` like this one.

**Comment by David-OConnor** [comment]
Fixed: Settings -> Rust -> Standard library -> Select the correct embedded toolchain (eg eabih)

**Comment by qiujiangkun** [comment]
Sometimes it happens to me as `Use of moved value. xxx: u32`

**Comment by tmbull** [comment]
@David-OConnor This was posted a long time ago but I hope you still might see this. Can you please provide a little more context around your suggested fix? I see the Standard library setting, but I'm not sure what the "correct embedded toolchain" is in my case or where I would find it... I'm using [avr-hal](https://github.com/Rahix/avr-hal) to program an ATMega328P (arduino) chip.
